src/data_conn.c

```diff
@@ -55,9 +55,13 @@ int replica_timeout = REPLICA_DEFAULT_TIMEOUT;
 		rcomm_cmd->resp_list[idx].io_resp_hdr.status =		\
 		    ZVOL_OP_STATUS_FAILED;				\
 		rcomm_cmd->resp_list[idx].data_ptr = NULL;		\
-		rcomm_cmd->resp_list[idx].status |= RECEIVED_ERR;	\
```

Comment:
this shouldn't cause any problem.. it may just delay the thread, until that gets timedout, which is waiting on this signal

Reply:
ok.. as I understood from you, setting 'status' can make rcomm_cmd get freed in other threads, and which is why this has been moved to the end.
Can you please add a comment for this, and, the precaution to be taken for further changes?